lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -424,7 +424,7 @@ def ipv4_address(vm)
     # Multiple reboots occur during guest customization in which a link-local
     # address is assigned. As such, we need to wait until a routable IP address
     # becomes available. This is most commonly an issue with Windows instances.
-    sleep 2 while vm.guest.net[bootstrap_nic_index].ipConfig.ipAddress.detect { |addr| IPAddr.new(addr.ipAddress).ipv4? }.origin == 'linklayer'
+    sleep 2 while (vm.guest.net[bootstrap_nic_index].ipConfig.ipAddress.detect { |addr| IPAddr.new(addr.ipAddress).ipv4? }.nil? || vm.guest.net[bootstrap_nic_index].ipConfig.ipAddress.detect { |addr| IPAddr.new(addr.ipAddress).ipv4? }.origin == 'linklayer')
```

Comment:
Let's drive some of this down to a method... What about something like

```
ORIGIN_IS_REAL_NIC = 'linklayer'
sleep 2 while vm_is_waiting_for_ip?(vm)
...
def vm_is_waiting_for_ip(vm)
  first_ip = vm.guest.net[bootstrap_nic_index].ipConfig.ipAddress.detect { |addr| IPAddr.new(addr.ipAddress).ipv4? }
  first_ip.nil? || first_ip.origin == ORIGIN_IS_REAL_NIC
end
```

Another approach would be to use `select` instead of `detect` and then use `empty?`...

What do you think of that?